relint.py

```diff
@@ -46,17 +52,73 @@ def load_config(path):
             )
 
 
-def lint_file(filename, tests):
-    try:
-        with open(filename) as fs:
-            content = fs.read()
-    except (IsADirectoryError, UnicodeDecodeError):
-        pass
-    else:
+def lint_file(filename, tests, diff=None):
+    if diff and diff.get(filename):
         for test in tests:
-            if any(fnmatch.fnmatch(filename, fp) for fp in test.filename):
+            for line_number, content in diff.get(filename).items():
                 for match in test.pattern.finditer(content):
-                    yield filename, test, match
+                    yield filename, test, match, line_number
+    else:
+        try:
+            with open(filename) as fs:
+                content = fs.read()
+        except (IsADirectoryError, UnicodeDecodeError):
+            pass
+        else:
+            for test in tests:
+                if any(fnmatch.fnmatch(filename, fp) for fp in test.filename):
+                    for match in test.pattern.finditer(content):
+                        yield filename, test, match, None
+
+
+def parse_diff(output):
+    changed_content = {}
+    current_file = None
+    line_after_number_line = False
+    current_line = None
+
+    pattern = re.compile(r"(([+]|[-])\d*[,]?\d*)")
+
+    for line in output.splitlines():
+        if line.startswith('diff'):
+            current_file = line[line.rfind(' b/')+3:]
+            line_after_number_line = False
+
+        elif line.startswith('@@'):
+            result = re.findall(pattern, line)
+            result = result[1][0].replace('+', '').split(',')
+            if result:
+                current_line = int(result[0])
+            line_after_number_line = True
+
+        elif line_after_number_line and line.startswith('+'):
+            if changed_content.get(current_file):
+                changed_content[current_file][current_line] = line[1:]
+            else:
+                changed_content[current_file] = {
+                    current_line: line[1:]
+                }
+            current_line += 1
+
+    return changed_content
```

Comment:
what is this a dict? Why don't you just parse the filenames and line numbers?

Reply:
It's a dict with filenames and line numbers...

Reply:
I would suggest to only return file name and line numbers. like so:
```
{
    'relint.py': [55],
}
```

Keep the rest of the bahavior as is.
https://github.com/codingjoe/relint/blob/master/relint.py#L84-L89
At this point in the code you can omit machtes that are not with this dictionary.